tests/scripts/restore-mon-quorum-using-osds.sh

```diff
@@ -0,0 +1,318 @@
+    kubectl -n "${NAMESPACE}" patch deployment "${mon}" -p '{"spec": {"template": {"spec": {"containers": [{"name": "mon", "command": ["sleep", "infinity"], "args": []}]}}}}'
+    kubectl -n "${NAMESPACE}" patch deployment "${mon}" --type=json --patch='[{"op":"remove", "path":"/spec/template/spec/containers/0/livenessProbe"}]'
+done
+
+wait_pods rook-ceph-mon "${DESIRED_MON_COUNT}" Running
+
+for osd in ${OSDS}; do
+    kubectl -n "${NAMESPACE}" patch deployment "${osd}" -p '{"spec": {"template": {"spec": {"containers": [{"name": "osd", "command": ["sleep", "infinity"], "args": []}]}}}}'
+    kubectl -n "${NAMESPACE}" patch deployment "${osd}" --type=json --patch='[{"op":"remove", "path":"/spec/template/spec/containers/0/livenessProbe"}]'
+done
+
+wait_pods rook-ceph-osd "${DESIRED_OSD_COUNT}" Running
+
+echo '[STEP3] done'
+
+echo '[STEP4] injecting mon-keyring to toolbox for the subsequent steps'
+
+kubectl -n "${NAMESPACE}" patch deployment "${TOOL_DEPLOYMENT}" --type=json \
+ --patch='[
+     {"op":"add", "path":"/spec/template/spec/volumes/-", "value": {"name": "mons-keyring", "secret": {"secretName": "rook-ceph-mons-keyring"}}},
+     {"op":"add", "path":"/spec/template/spec/containers/0/volumeMounts/-", "value": {"name": "mons-keyring", "mountPath": "/keyrings"}}
+ ]'
+
+kubectl -n "${NAMESPACE}" delete pod -l app=rook-ceph-tools
+wait_pods rook-ceph-tools 1 Running
+
+echo '[STEP4] done'
+
+echo '[STEP5] gathering the information needed to re-generate the monitor store'
+
+# we should disable -e flag because some OSDs are down
+for i in ${OSD_IDS}; do
+    echo "Gather information from osd $i"
+    name=$(kubectl get -n "${NAMESPACE}" pod -l app=rook-ceph-osd -l osd="${i}" -o json | jq -r '.items[].metadata.name')
+    kubectl exec -i -n "${NAMESPACE}" "${name}" -- rm -rf "${TMP_MON_STORE}"
+    kubectl exec -i -n "${NAMESPACE}" "${name}" -- mkdir -p "${TMP_MON_STORE}"
+    tar cf - -C "${TMP_MON_STORE}" . | kubectl exec -i -n "${NAMESPACE}" "${name}" -- tar xf - -C "${TMP_MON_STORE}"
+    kubectl exec -i -n "${NAMESPACE}" "${name}" -- ceph-objectstore-tool --data-path /var/lib/ceph/osd/ceph-"${i}" --no-mon-config --op update-mon-db --mon-store-path "${TMP_MON_STORE}"
+    rm -rf "${TMP_MON_STORE}"/*
+    kubectl exec -i -n "${NAMESPACE}" "${name}" -- tar cf - -C "${TMP_MON_STORE}" . | tar xf - -C "${TMP_MON_STORE}"
+done
+
+echo '[STEP5] done'
+
+echo '[STEP6] re-generating monitor store'
+
+TOOL=$(kubectl get -n "${NAMESPACE}" po -l app=rook-ceph-tools -o json | jq -r '.items[0].metadata.name')
+kubectl exec -i -n "${NAMESPACE}" "${TOOL}" -- rm -rf "${TMP_MON_STORE}"
+kubectl exec -i -n "${NAMESPACE}" "${TOOL}" -- mkdir -p "${TMP_MON_STORE}"
+tar cf - -C "${TMP_MON_STORE}" . | kubectl exec -i -n "${NAMESPACE}" "${TOOL}" -- tar xf - -C "${TMP_MON_STORE}"
+kubectl exec -i -n "${NAMESPACE}" "${TOOL}" -- ceph-monstore-tool --fsid="${FSID}" "${TMP_MON_STORE}" rebuild -- --keyring /keyrings/keyring --mon-ids ${MON_IDS}
```

Comment:
What if the mon deployments don't exist? Do we require they exist?

Reply:
Yes, I'll fix it. We might rescue non-mon deployment case. However, I'd like to make it out of scope for now.